model/map-ldap/src/main/java/org/keycloak/models/map/storage/ldap/condition/NoopCondition.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2022. Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.models.map.storage.ldap.condition;
+
+import org.keycloak.storage.ldap.idm.query.Condition;
+
+/**
+ * @author Alexander Schwartz
+ */
+public class NoopCondition implements Condition {
```

Comment:
That depends. It will always prune the AST. 

Some examples: 
* AND(NOOP, X) is And(X), 
* AND(NOOP) is NOOP.
* OR(X, NOOP) is OR (X)
* OR(NOOP) is NOOP
* NOT(NOOP) is NOOP